source/includes/api/_user_management.md

```diff
@@ -1,47 +1,75 @@
+Use `Content-Type: application/json` if you are calling the endpoint from an application, and would like to get a JSON formatted response. 
+If you are accessing these endpoints from a browser, it will typically generate a request with the `Content-Type: text/html` header, in which case the reply will be in HTML format.
 
-Unlike the other parts of the API, this section covers endpoints that provide end-user interaction, rendering HTML pages or sending emails to reset passwords and such. Some elements of these interfaces can be configured to match specific projects identities (RW, GFW, etc). To specify which project your requests come from, you can add an optional `origin` query parameter to your requests, with the name of the application. If matching visual elements exist, they will be used in the resulting interfaces displayed to the user.
+Keep in mind that not all endpoints support both formats, and will output either HTML or JSON, no matter which `Content-Type` value you provide.
 
+## General notes on RW API users
+
+User accounts in the RW API work pretty much like you would expect: you create a user account, login with it, and use it on your day-to-day interactions to access certain actions and associate certain RW API resources with it.
+
+Besides authentication, user accounts are also used for authorization. Authorization is implemented on a per-service basis, but is commonly built on top of 3 elements associated with your a user account:
+
+- `id`
+- `role`
+- `application`
+
+`id` is a unique identifier of your account. Certain actions may be limited to specific users, typically in the context of a given resource being associated directly with a specific user `id`.
+
+`role` can be one of 3 values: `USER`, `MANAGER` and `ADMIN`. While not required nor enforced, typically they are used hierarchically in that order, from least to most permissive. A common pattern you'll find on some services is: `USER` accounts can only create new resources, `MANAGER` accounts can create new resources, and edit or delete resources created by them, while `ADMIN` accounts can do all of the above even for resources created by other users.
+
+`application` is a list of keys meant to identify the different client applications that are built using the RW API. It's present not only on user accounts, but also on many of the resources found on the RW API, either as a single value or as a list of values. Typically, in order to manipulate a given resource, that resource and the user account must have at least one overlapping `application` value.
+
+Keep in mind that it's up to each individual RW API service (dataset, widget, layer, etc) to define how they restrict or allow actions based on these or other factors, so the examples above may not be true for all cases. Refer to the documentation of each resource and endpoint for more details on restrictions they may have regarding user accounts and their properties.    
```

Comment:
On a scale of 1-10, how good do we feel about how comprehensive the role restriction documentation is throughout these docs?
Separately, consistent would you say the definitions of user, manager, and admin are?

Reply:
By "throughout these docs" I assume you mean the docs as a whole, not just this section. Based on what I've seen (and documented) so far, the core elements (dataset, layer, widget, metadata) see to be consistent with the role description provided here. As for other microservices, like area or subscription, that also have a CRUD base, I am less familiar with, so I can't really say - maybe @ikas knows more than I do, as he's been working on those.

I assume you are trying to evaluate if we can havea single definition of the permissions of each role, that then applies to the API as a whole, correct? At this point, I feel like we don't know enough about several MS to build that reliably. Perhaps that's something we can tackle specifically in a future task, and ideally even make that standard part of the dev guidelines, so future development respects that.

Reply:
Okay thanks. I'll consider including this in the next contract.